hawkbit-dmf-parent/hawkbit-dmf-rabbitmq-test/src/main/java/org/eclipse/hawkbit/rabbitmq/test/AbstractAmqpIntegrationTest.java

```diff
@@ -67,4 +72,26 @@ protected Message createMessage(final Object payload, final MessageProperties me
         return getDmfClient().getMessageConverter().toMessage(payload, messageProperties);
     }
 
+    protected int getQueueMessageCount(String queueName) {
+        return Integer
+                .parseInt(rabbitAdmin.getQueueProperties(queueName).get(RabbitAdmin.QUEUE_MESSAGE_COUNT).toString());
+    }
+
+    protected RabbitAdmin getRabbitAdmin() {
+        return rabbitAdmin;
+    }
+
+    RabbitTemplate createDmfClient() {
+        final RabbitTemplate template = new RabbitTemplate(connectionFactory);
+        template.setMessageConverter(new Jackson2JsonMessageConverter());
+        template.setReceiveTimeout(TimeUnit.SECONDS.toMillis(3));
+        template.setReplyTimeout(TimeUnit.SECONDS.toMillis(3));
+        template.setExchange(getExchange());
+        return template;
+    }
+
+    protected ConnectionFactory getConnectionFactory() {
```

Comment:
Do not expose the hole `ConnectionFactory` here when the virtual host is sufficient